### PR TITLE
feat!: support data connectors

### PR DESCRIPTION
**Pull request by leafty**
Add support for mounting data connectors (instead of `storages_v2`) with Renku 2.0 sessions.

/deploy renku-data-services=leafty/feat-namespaced-data-connectors-5 renku-ui=ciyer/data-connector-secrets

### CODE REVIEW

File: renku_notebooks/api/notebooks.py [hunk 1]
```diff
@@ -270,7 +270,8 @@ def renku_2_launch_notebook_helper(
         default_url=default_url,
         lfs_auto_fetch=lfs_auto_fetch,
         cloudstorage=cloudstorage,
-        cloudstorage_endpoint="storages_v2",
+        # TODO: update here
```
Comment (leafty) [inline renku_notebooks/api/notebooks.py:273]:
```suggestion
```

**Review by olevski — APPROVED** [review]

**Comment by olevski** [comment]
@leafty please do note merge this into `master` on notebooks until the other code from data-service is ready.

**Comment by olevski** [comment]
@leafty you can merge this. You go first. I will merge my caching PR after you. Then I will make a release of notebooks.

**Comment by leafty** [comment]
> @leafty you can merge this. You go first. I will merge my caching PR after you. Then I will make a release of notebooks.

I was intending to merge this only once the whole data connectors feature is ready so that components are released in sync.

**Pull request merged**